runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/WindowDoFnOperator.java

```diff
@@ -288,6 +384,43 @@ public int compare(
         watermarkTimersQueue.add(keyedTimer);
       }
     }
+
+    int numProcessingTimeTimers = dataIn.readInt();
+
+    processingTimeTimers = new HashSet<>(numProcessingTimeTimers);
+    processingTimeTimersQueue = new PriorityQueue<>(
+        Math.max(numProcessingTimeTimers, 1),
+        new Comparator<Tuple2<ByteBuffer, TimerInternals.TimerData>>() {
+          @Override
+          public int compare(
+              Tuple2<ByteBuffer, TimerInternals.TimerData> o1,
+              Tuple2<ByteBuffer, TimerInternals.TimerData> o2) {
+            return o1.f1.compareTo(o2.f1);
+          }
+        });
+
+    processingTimeTimerTimestamps = HashMultiset.create();
+
+    for (int i = 0; i < numProcessingTimeTimers; i++) {
+      int length = dataIn.readInt();
+      byte[] keyBytes = new byte[length];
+      dataIn.readFully(keyBytes);
+      TimerInternals.TimerData timerData = timerCoder.decode(dataIn, Coder.Context.NESTED);
+      Tuple2<ByteBuffer, TimerInternals.TimerData> keyedTimer =
+          new Tuple2<>(ByteBuffer.wrap(keyBytes), timerData);
+      if (processingTimeTimers.add(keyedTimer)) {
+        processingTimeTimersQueue.add(keyedTimer);
+
+        //If this is the first timer added for this timestamp register a timer Task
+        if (processingTimeTimerTimestamps.add(timerData.getTimestamp().getMillis(), 1) == 0) {
+          // this registers a timer with the Flink processing-time service
+          ScheduledFuture<?> scheduledFuture =
+              registerTimer(timerData.getTimestamp().getMillis(), this);
+          processingTimeTimerFutures.put(timerData.getTimestamp().getMillis(), scheduledFuture);
```

Comment:
I believe `restoreState(...)` is called before `open(..)` which means that `processingTimeTimerFutures` will be null and cause a NullPointerException.